lib/validators/email_validator.rb

```diff
@@ -16,9 +16,13 @@ def validate_each(record, attribute, value)
   end
 
   def email_in_restriction_setting?(setting, value)
-    domains = setting.gsub('.', '\.')
-    regexp = Regexp.new("@(#{domains})", true)
-    value =~ regexp
+    is_blacklisted = false
+    domains = setting.split("|")
+    email_domain = value.split("@").last if value
+    domains.each { |domain|
+      is_blacklisted = true if email_domain && (email_domain == domain || email_domain.end_with?(".#{domain}"))
+    }
+    is_blacklisted
```

Comment:
Hmm, I would have simply updated the regex to something like `Regexp.new("@(.+\.)?(#{domains})", true)`